static/app/views/issueList/groupStatsProvider.tsx

```diff
@@ -0,0 +1,129 @@
+import {createContext, useContext} from 'react';
+import {dropUndefinedKeys} from '@sentry/utils';
+import * as reactQuery from '@tanstack/react-query';
+
+import {ApiResult} from 'sentry/api';
+import type {Group, GroupStats, Organization, PageFilters} from 'sentry/types';
+import {getUtcDateString} from 'sentry/utils/dates';
+import {UseQueryResult} from 'sentry/utils/queryClient';
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+function getEndpointParams(
+  p: Pick<GroupStatsProviderProps, 'selection' | 'period' | 'query' | 'groupIds'>
+): StatEndpointParams {
+  const params: StatEndpointParams = {
+    project: p.selection.projects,
+    environment: p.selection.environments,
+    groupStatsPeriod: p.period,
+    query: p.query,
+    groups: p.groupIds,
+    ...p.selection.datetime,
+  };
+
+  if (p.selection.datetime.period) {
+    delete params.period;
+    params.statsPeriod = p.selection.datetime.period;
+  }
+  if (params.end) {
+    params.end = getUtcDateString(params.end);
+  }
+  if (params.start) {
+    params.start = getUtcDateString(params.start);
+  }
+
+  return dropUndefinedKeys(params);
+}
+
+const GroupStatsContext = createContext<UseQueryResult<
+  Record<string, GroupStats>
+> | null>(null);
+
+export function useGroupStats(groupId: Group['id']): GroupStats;
+export function useGroupStats(): UseQueryResult<Record<string, GroupStats>>;
+export function useGroupStats(maybeGroupId?: Group['id']) {
```

Comment:
Open to suggestions here, I overloaded the stats hook to either give us the entire stats response or just the data of a single group (pending a provided groupId arg). Returning a more complete object that resembled rtk value would probably be better here so that the callers have access to loading, error or any of the other state. Very much open to suggestions on this.